godot-core/src/lib.rs

```diff
@@ -77,14 +79,41 @@ pub mod private {
         F: FnOnce() -> R + std::panic::UnwindSafe,
         S: std::fmt::Display,
     {
-        match std::panic::catch_unwind(code) {
+        let location_line: Arc<Mutex<u32>> = Arc::new(Mutex::new(0));
+        let location_file: Arc<Mutex<String>> = Arc::new(Mutex::new(String::from("file")));
+
+        // Back up previous hook, set new one
+        let prev_hook = std::panic::take_hook();
+        {
+            let location_line = location_line.clone();
+            let location_file = location_file.clone();
+            std::panic::set_hook(Box::new(move |panic_info| {
+                if let Some(location) = panic_info.location() {
+                    *location_line.lock().unwrap() = location.line();
+                    *location_file.lock().unwrap() = String::from(location.file());
+                } else {
+                    println!("panic occurred but can't get location information...");
+                }
+            }));
+        }
+
+        // Run code that should panic, restore hook
+        let panic = std::panic::catch_unwind(code);
+        std::panic::set_hook(prev_hook);
+
+        match panic {
             Ok(result) => Some(result),
             Err(err) => {
                 // Flush, to make sure previous Rust output (e.g. test announcement, or debug prints during app) have been printed
                 // TODO write custom panic handler and move this there, before panic backtrace printing
                 flush_stdout();
 
-                log::godot_error!("Rust function panicked. Context: {}", error_context());
+                log::godot_error!(
+                    "Rust function panicked at file {} at line {}. Context: {}",
```

Comment:
Should we maybe include column information here as well? It seems like it's pretty easy to get from the PanicInfo and is useful for debugging.

Also a minor thing but i feel like it should be "Rust function panicked *in* file {} at line {}" instead.

Reply:
I would not add the column info, it adds noise and makes parsing the line number slower.

With message _and_ line, it's never really ambiguous. Hardly any Rust line has more than one panic, especially not when using rustfmt...

Reply:
fair, but keep in mind that normal rust panics do print out column info:
```
thread 'main' panicked at '', src/main.rs:2:5
```
code:
```rs
fn main() {
    panic!("")
}
```

Reply:
Yeah, but this is mostly relevant for tooling, like IDEs/debuggers that can directly jump to the panic or highlight it.